presto-main/src/main/java/io/prestosql/sql/analyzer/Analysis.java

```diff
@@ -139,6 +139,9 @@
     private final Multiset<RowFilterScopeEntry> rowFilterScopes = HashMultiset.create();
     private final Map<NodeRef<Table>, List<Expression>> rowFilters = new LinkedHashMap<>();
 
+    private final Multiset<ColumnMaskScopeEntry> columnMaskScopes = HashMultiset.create();
+    private final Map<NodeRef<Table>, Map<String, List<Expression>>> columnMasks = new LinkedHashMap<>();
```

Comment:
if row filters and column masks are using same mechanism during semantic analysis, should we extract common code as some entity?

Reply:
I don't think that saves us much and introduces unnecessary abstraction. We'd need a generic holder class parameterized on the scope entry type and the container (`List<Expression>` vs `Map<String, List<Expression>`). We also still have to have the register/unregister/hasXXX/getXXX methods, the logic to create the keys, etc., in Analysis.